### PR TITLE
[RDY] API: Adding cterm support to nvim_set_hl

### PR DESCRIPTION
**Pull request by shadmansaleh**
Currently it's not possible to add cterm highlight with nvim_set_hl() this pr adds that . It's a non breaking change .
It adds optional ctermfg, ctermbg and cterm keys to color dictonary argument.

The idea credit goes to bfredl

### CODE REVIEW

**Comment by shadmansaleh** [comment]
Are describe blocks environment isolated from each other ?

File: test/functional/api/highlight_spec.lua [hunk 1]
```diff
@@ -158,3 +158,97 @@ describe('API: highlight',function()
...
+  local highlight3_result_cterm = {
+    background = highlight_color.ctermbg,
+    foreground = highlight_color.ctermfg,
+    italic = true,
+    reverse = true,
+    undercurl = true,
+  }
+
+  local function get_ns()
+    local ns = meths.get_namespaces().Test_set_hl
```
Comment (bfredl) [inline test/functional/api/highlight_spec.lua:221]: you don't need this check, just use `meths.create_namespace' directly.

**Comment by bfredl** [comment]
@shadmansaleh if you mean things like before_each/after_each, yes.

File: test/functional/api/highlight_spec.lua [hunk 2]
```diff
@@ -158,3 +158,97 @@ describe('API: highlight',function()
...
+    underline = true,
+  }
+  local highlight3_result_cterm = {
+    background = highlight_color.ctermbg,
+    foreground = highlight_color.ctermfg,
+    italic = true,
+    reverse = true,
+    undercurl = true,
+  }
+
```
Comment (bfredl) [inline test/functional/api/highlight_spec.lua:219]: you probably want `before_each(clear)`

File: test/functional/api/highlight_spec.lua (hunk 1 above)
Comment (shadmansaleh) [reply test/functional/api/highlight_spec.lua]: Done . It seems that check is alreadya  part of create_namespace

File: test/functional/api/highlight_spec.lua (hunk 2 above)
Comment (shadmansaleh) [reply test/functional/api/highlight_spec.lua:219]: I've added it but I'm not sure what it does . Does it call module.clear() in tests/functional/helpers.lua ?

Is it suppose to clear the namespace ?
Actually nvim_buf_clear_namespcae clears highlights of that namespace in  the buffer . But is there anything equavelent of :hi clear for namespace (removing highlight definations too not just applications )

**Comment by shadmansaleh** [comment]
> @shadmansaleh if you mean things like before_each/after_each, yes.

I wasn't meaning before/after_each . like in the highlight_spec.lua there are two descrive blocks one testing highlight retrival and one testing highlight setting . Are those two isolatedfrom each other . What my concern was if I create a namespace and don't remove it(I couldn't find a way to do so) will it have any chance of messing up a test down the line .

**Comment by bfredl** [comment]
they are isolated if you use `before_each(clear)` :)

File: test/functional/api/highlight_spec.lua (hunk 2 above)
Comment (bfredl) [reply test/functional/api/highlight_spec.lua:219]: it restarts neovim so all state is cleared.

**Comment by shadmansaleh** [comment]
Thanks for confirming . Is currently any way to remove a namespace ?

**Comment by bfredl** [comment]
no and I don't think there should be. however we should add the possibility to _clear_ all highlights in a namespace.

**Comment by shadmansaleh** [comment]
> however we should add the possibility to clear all highlights in a namespace.

That should be enough .

I don't think there is a way to list highlight_groups under a namespace either . is there ?

**Comment by shadmansaleh** [comment]
@bfredl any progress here ?

**Pull request merged**